modin/core/execution/dask/implementations/pandas_on_dask/partitioning/virtual_partition.py

```diff
@@ -36,23 +38,97 @@ class PandasOnDaskDataframeAxisPartition(PandasDataframeAxisPartition):
         Whether to get node IP addresses of conforming partitions or not.
     full_axis : bool, default: True
         Whether or not the virtual partition encompasses the whole axis.
+    call_queue : list, optional
+        A list of tuples (callable, args, kwargs) that contains deferred calls.
     """
 
-    def __init__(self, list_of_blocks, get_ip=False, full_axis=True):
-        if not full_axis:
-            raise NotImplementedError(
-                "Pandas on Dask execution requires full-axis partitions."
+    axis = None
+
+    def __init__(self, list_of_blocks, get_ip=False, full_axis=True, call_queue=None):
+        self.call_queue = call_queue or []
+        self.full_axis = full_axis
+        if isinstance(list_of_blocks, PandasOnDaskDataframePartition):
+            list_of_blocks = [list_of_blocks]
+        if not any(
+            isinstance(o, PandasOnDaskDataframeVirtualPartition) for o in list_of_blocks
+        ):
+            self.list_of_partitions_to_combine = list_of_blocks
+            return
+
+        assert (
```

Comment:
Can we the ray virtual partition constructor comments, e.g. [this](https://github.com/modin-project/modin/blob/0ab04e104a60419c464fc1833a13f7f2b92e7983/modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py#L61), explaining some of the non-trivial code?

Reply:
Sure, that makes sense to me!